driver-core/src/main/java/com/datastax/driver/core/MaterializedViewMetadata.java

```diff
@@ -101,7 +101,15 @@ static MaterializedViewMetadata build(KeyspaceMetadata keyspace, Row row, Map<St
         // 'SELECT * FROM ...'
         Set<ColumnMetadata> otherColumns = new TreeSet<ColumnMetadata>(columnMetadataComparator);
         for (ColumnMetadata.Raw rawCol : rawCols.values()) {
-            ColumnMetadata col = ColumnMetadata.fromRaw(view, rawCol);
+            DataType dataType;
+            if(cassandraVersion.getMajor() >= 3) {
+                dataType = DataTypeParser.parse(rawCol.dataType, cluster.getMetadata(), keyspace.userTypes, false);
```

Comment:
Data types are now resolved later in the process, because we need user types to be resolved first.